conans/client/cmd/uploader.py

```diff
@@ -501,31 +509,42 @@ def compress_files(files, symlinks, name, dest_dir, output=None):
         # tgz_contents = BytesIO()
         tgz = gzopen_without_timestamps(name, mode="w", fileobj=tgz_handle)
 
-        for filename, dest in sorted(symlinks.items()):
-            info = tarfile.TarInfo(name=filename)
+        for relpath, dest in sorted(gathered_files.symlinks.items()):
+            if dest.startswith("."):
+                warn_outside_sources(os.path.join(gathered_files.base_folder, relpath), dest)
+
+            info = tarfile.TarInfo(name=relpath)
             info.type = tarfile.SYMTYPE
             info.linkname = dest
             info.size = 0  # A symlink shouldn't have size
             tgz.addfile(tarinfo=info)
 
         mask = ~(stat.S_IWOTH | stat.S_IWGRP)
         i_file = 0
-        n_files = len(files)
+        n_files = len(gathered_files.files)
         last_progress = None
         if output and n_files > 1 and not output.is_terminal:
             output.write("[")
-        for filename, abs_path in sorted(files.items()):
+
+        for filename, abs_path in sorted(gathered_files.files.items()):
             info = tarfile.TarInfo(name=filename)
             info.size = os.stat(abs_path).st_size
             info.mode = os.stat(abs_path).st_mode & mask
             if os.path.islink(abs_path):
+                linkto = os.path.join(os.path.dirname(abs_path), os.readlink(abs_path))
+                linkto = os.path.normpath(linkto)
+                linkto_rel = os.path.relpath(linkto, gathered_files.base_folder)
+                if linkto_rel.startswith("."):
+                    warn_outside_sources(abs_path, linkto)
+
                 info.type = tarfile.SYMTYPE
                 info.size = 0  # A symlink shouldn't have size
-                info.linkname = os.readlink(abs_path)  # @UndefinedVariable
+                info.linkname = linkto_rel  # @UndefinedVariable
```

Comment:
⚠️ [Breaking?] Using the relative path instead of the one returned by `os.readlink`. This doesn't affect Conan behavior because these symlinks pointing outside the zipped folder would be removed in the `tar_extract` function, but the zipped file contents will be different.

Reply:
yes, I think it's breaking, as long as someone uses `readlink` to get information about symbolic link and relies on it.